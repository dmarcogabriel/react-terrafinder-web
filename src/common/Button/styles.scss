@import '../../styles';

.button {
  background: $primary-dark;
  color: $white;
  padding: $size-sm;
  border-radius: $size-xs;
  box-shadow: 0 7px 32px 2px rgba(0, 0, 0, 0.75);
  cursor: pointer
}
