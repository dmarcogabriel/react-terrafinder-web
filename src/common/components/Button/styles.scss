@import '../../../styles';

.button {
  background: $primary-dark;
  color: $white;
  padding: $size-sm;
  border-radius: 0.25rem;
  box-shadow: $shadow-sm;
  cursor: pointer;
}
