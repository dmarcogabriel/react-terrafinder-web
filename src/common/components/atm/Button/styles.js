import styled, { css } from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';

const modifierConfig = {
  default: ({ theme: { color } }) => css`
    background: ${color.white.df};
    color: ${color.black.df};
  `,
  primary: ({ theme: { color } }) => css`
    background: ${color.primary.df};
    color: ${color.white.df};
  `,
  secondary: ({ theme: { color } }) => css`
    background: ${color.white.df};
    color: ${color.blue.df};
  `,
  success: ({ theme: { color } }) => css`
    background: ${color.green.df};
    color: ${color.white.df};
  `,
  // todo: fix rounded button
  rounded: () => css`
    /* border-radius: 20%; */
  `,
  // todo: add a disabled style
  disabled: () => css`
    /*  */
  `,
};

export const ButtonBase = styled.button.attrs({ type: 'button' })`
  cursor: pointer;
  text-align: center;
  ${({ theme }) => css`
    font-family: ${theme.fontFamily.roboto};
    box-shadow: ${theme.shadow};
    padding: ${theme.size.sm};
    border-radius: ${theme.size.borderRadius};
  `}

  ${applyStyleModifiers(modifierConfig)}
`;
