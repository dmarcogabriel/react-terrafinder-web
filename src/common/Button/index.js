import React from 'react';
import './styles.scss';

export default function Button({ className, children, onClick, ...props }) {
  return (
    <button className="button" type="button" onClick={onClick} {...props}>
      {children}
    </button>
  );
}
