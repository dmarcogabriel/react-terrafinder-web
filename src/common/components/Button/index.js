import React from 'react';
import './styles.scss';
import cn from 'classnames';

export default function Button({ children, onClick, className, dataTestId }) {
  return (
    <button
      data-testid={dataTestId || 'button'}
      className={cn('button', className)}
      type="button"
      onClick={onClick}
    >
      {children}
    </button>
  );
}
