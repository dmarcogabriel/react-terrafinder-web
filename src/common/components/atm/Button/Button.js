import React from 'react';
import _ from 'lodash';
import { ButtonBase } from './styles';

export const Button = ({
  children,
  onClick,
  dataTestId,
  modifiers = [],
  ...props
}) => (
  <ButtonBase
    modifiers={_.isEmpty(modifiers) ? ['default'] : modifiers}
    data-testid={dataTestId || 'button'}
    onClick={onClick}
    {...props}
  >
    {children}
  </ButtonBase>
);
