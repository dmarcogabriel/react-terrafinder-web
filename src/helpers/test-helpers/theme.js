import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/themes';

export const withTheme = (component) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);

export const renderWithTheme = (component) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
