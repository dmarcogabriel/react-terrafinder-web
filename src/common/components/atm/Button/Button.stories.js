import React from 'react';
import { Button as CommonButton } from './Button';

export default {
  title: 'Common/Button',
  component: CommonButton,
  argTypes: {},
};

const Template = (args) => <CommonButton {...args} />;

export const Default = Template.bind();
Default.args = {
  children: 'This is a Button',
};

export const Primary = Template.bind();
Primary.args = {
  children: 'This is a Primary Button',
  modifiers: ['primary'],
};

export const Secondary = Template.bind();
Secondary.args = {
  children: 'This is a Secondary Button',
  modifiers: ['secondary'],
};

export const Success = Template.bind();
Success.args = {
  children: 'This is a Success Button',
  modifiers: ['success'],
};
