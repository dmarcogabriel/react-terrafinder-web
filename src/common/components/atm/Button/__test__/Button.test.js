import React from 'react';
import ReactDOM from 'react-dom';
import { renderWithTheme, withTheme } from 'helpers/test-helpers/theme';
import Button from '..';

describe('atm/Button', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(withTheme(<Button />), div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // todo: add tests here...

  it('matches snapshot', () => {
    const tree = renderWithTheme(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
