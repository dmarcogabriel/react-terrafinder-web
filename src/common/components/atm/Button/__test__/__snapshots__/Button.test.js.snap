// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atm/Button matches snapshot 1`] = `
.c0 {
  cursor: pointer;
  text-align: center;
  font-family: Roboto;
  box-shadow: 0 0.2rem 0.6rem 0 rgba(0,0,0,0.7);
  padding: 0.75rem;
  border-radius: 0.25rem;
  background: #fff;
  color: #333;
}

<button
  className="c0"
  data-testid="button"
  type="button"
/>
`;
